--- hdf5_HL_REGION-1.1.3/Makefile.orig	2015-01-28 21:43:03.733570959 +0100
+++ hdf5_HL_REGION-1.1.3/Makefile	2015-01-28 21:43:08.200237439 +0100
@@ -70,17 +70,17 @@
 	 cd ../test; $(MAKE));fi
 
 install:
-	(mkdir -p $(PREFIX)/lib)
-	(mkdir -p $(PREFIX)/include)
-	(cp src/libhdf5_hl_region.a $(PREFIX)/lib )
-	(cp src/hl_region_H5LRpublic.h $(PREFIX)/include )
-	(cp src/hl_region_H5LTpublic.h $(PREFIX)/include )
-	(cp src/h5hl_region.h $(PREFIX)/include )
-	(cp src/h5hl_api.h $(PREFIX)/include )
+	(mkdir -p $(DESTDIR)$(LIBDIR))
+	(mkdir -p $(DESTDIR)$(PREFIX)/include)
+	($(MAKE) -C src install)
+	(cp src/hl_region_H5LRpublic.h $(DESTDIR)$(PREFIX)/include )
+	(cp src/hl_region_H5LTpublic.h $(DESTDIR)$(PREFIX)/include )
+	(cp src/h5hl_region.h $(DESTDIR)$(PREFIX)/include )
+	(cp src/h5hl_api.h $(DESTDIR)$(PREFIX)/include )
 # if fortran enabled then compile FORTRAN source
-	if test -e fortran/src/h5hl_region.o ; then \
-	(cp fortran/src/libhdf5_hl_region_fortran.a $(PREFIX)/lib; \
-	 cp fortran/src/*.mod $(PREFIX)/include );fi
+	if test -e fortran/src/h5hl_region.lo ; then \
+	($(MAKE) -C fortran/src install; \
+	 cp fortran/src/*.mod $(DESTDIR)$(PREFIX)/include );fi
 
 tests:
 	(cd test; $(MAKE) tests )
